services/scabbard/libscabbard/src/store/scabbard_store/commit.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use splinter::error::InvalidStateError;
+use splinter::service::FullyQualifiedServiceId;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct CommitEntry {
+    service_id: FullyQualifiedServiceId,
+    epoch: i64,
+    value: String,
+    decision: Option<ConsensusDecision>,
+}
+
+#[allow(dead_code)]
```

Comment:
We will have to be careful to make sure none of these are left behind once the structs are used